lua/entities/gmod_wire_expression2/core/entity.lua

```diff
@@ -526,16 +526,13 @@ end
 --- Sets <this>'s skin number.
 e2function void entity:setSkin(skinIndex)
 	if not IsValid(this) then return self:throw("Invalid entity!", nil) end
-	if this:IsPlayer() then return self:throw("You cannot set the skin of a player!", nil) end
+	if not isOwner(self, this) then return self:throw("You do not own this entity!", nil) end
+
+	if this:SkinCount() < 1 then return end
```

Comment:
Might as well add a `self:throw()` here now that you've separated it: would help avoid confusion
```suggestion
	if this:SkinCount() < 1 then return self:throw("This entity does not have any skins", nil) end
```

Reply:
In my opinion it shouldn't throw if you're out of bounds. It seems like too much.

Reply:
@Vurv78 I agree with @Denneisk - I think an invalid skin index shouldn't do anything, or should reset skin to default. What do you think?

Reply:
Yeah @AshleyJamesy agree, just reset skin to default would be better

Reply:
Most of the time the setSkin function is used for player model cosmetics

Reply:
Throwing an error on `@strict` is a good idea, but I agree with Ashley that passing invalid data should trigger a reset to defaults in the same way functions like `setMaterial` do.